specs/networking/node-identification.md

```diff
@@ -0,0 +1,32 @@
+ETH 2.0 Networking Spec - Node Identification
+===
+
+# Abstract
+
+This specification describes how Ethereum 2.0 nodes identify and address each other on the network.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL", NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+# Specification
+
+Clients use Ethereum Node Records (as described in [EIP-778](http://eips.ethereum.org/EIPS/eip-778)) to discover one another. Each ENR includes, among other things, the following keys:
+
+- The node's IP.
+- The node's TCP port.
+- The node's public key.
+
+For clients to be addressable, their ENR responses MUST contain all of the above keys. Client MUST verify the signature of any received ENRs, and disconnect from peers whose ENR signatures are invalid. Each node's public key MUST be unique.
+
+The keys above are enough to construct a [multiaddr](https://github.com/multiformats/multiaddr) for use with the rest of the `libp2p` stack.
```

Comment:
libp2p peer IDs are derived from the public key protobuf, which is just key type + bytes. Here's the spec: https://github.com/libp2p/specs/pull/100. Both SECIO and TLS 1.3 validate peer IDs against the pubkey, so following the spec is important or connections will fail.

Reply:
As I mention in https://github.com/libp2p/specs/pull/100/files#r266291995 - protobuf is not deterministic, and thus not great for feeding into a hashing function to determine an ID, unless you used a modified protobuf version that's locked down.

Reply:
Wouldn't this be handled at the `libp2p` layer? Here we're describing how to construct a `multiaddr` from an ENR; the actual handling of the `multiaddr` itself and the underlying hash construction would be the responsibiliy of `libp2p`.

Reply:
it would, but libp2p itself looks broken in this case - we need to keep an eye on that upstream issue so that we don't spread the breakage further.

Does using ENR require decoding RLP in this context?